assembly/std/index.ts

```diff
@@ -443,46 +582,64 @@ export function fileToByteArray(
 }
 
 /**
- * Returns the current period
+ * Retrieves the current period of the network.
+ *
+ * @returns the current period.
+ *
  */
 export function currentPeriod(): u64 {
   return env.currentPeriod();
 }
 
 /**
- * Returns the current thread
+ * Retrieves the current thread of the execution context.
+ *
+ * @returns the current thread.
+ *
  */
 export function currentThread(): u8 {
   return env.currentThread();
 }
 
 /**
- * Constructs an event given a key and arguments
+ * Constructs a pretty formatted event with given key and arguments.
+ *
+ * @remarks
+ * The result is meant to be used with the {@link generateEvent} function.
+ * It is useful to generate events from an array.
+ *
+ * @param key - the string event key.
+ *
+ * @param args - the string array arguments.
  *
- * @see {@link generateEvent}
+ * @returns the stringified event.
  *
- * @param key - event key
- * @param args - array of string arguments.
- * @returns stringified event.
  */
 export function createEvent(key: string, args: Array<string>): string {
   return `${key}:`.concat(args.join(','));
 }
 
 /**
- * Computing the sha256 of the passed parameter and return the hash as a byte array.
+ * Computes the SHA256 hash of the given `data`.
+ *
+ * @remarks
+ * The SHA256 hash algorithm produces a 32-byte hash, which is returned as a `StaticArray<u8>`.
+ *
+ * @param data - The data to hash.
+ *
+ * @returns The SHA256 hash of the `data`, serialized as a `StaticArray<u8>`.
  *
- * @param bytecode - StaticArray<u8>
- * @returns - Computed Sha256 in StaticArray<u8>
  */
-export function sha256(bytecode: StaticArray<u8>): StaticArray<u8> {
-  return env.sha256(bytecode);
+export function sha256(data: StaticArray<u8>): StaticArray<u8> {
```

Comment:
Is this function fully generic ?
Even if so, was it built to answer a specific use case to describe to users ?
Such as computing the hash of a contract ?